From c108c54011f6256806d4fe1a653e631fa66dba51 Mon Sep 17 00:00:00 2001
From: Kito Cheng <kito.cheng@sifive.com>
Date: Wed, 19 May 2021 18:06:27 +0800
Subject: [PATCH 01/15] testuite: Check pthread for omp module testing

gcc/testsuite/ChangeLog:

	* g++.dg/modules/omp-1_a.C: Check pthread is available.
	* g++.dg/modules/omp-1_b.C: Ditto.
	* g++.dg/modules/omp-1_c.C: Ditto.
	* g++.dg/modules/omp-2_a.C: Ditto.
	* g++.dg/modules/omp-2_b.C: Ditto.
---
 gcc/testsuite/g++.dg/modules/omp-1_a.C | 1 +
 gcc/testsuite/g++.dg/modules/omp-1_b.C | 1 +
 gcc/testsuite/g++.dg/modules/omp-1_c.C | 1 +
 gcc/testsuite/g++.dg/modules/omp-2_a.C | 1 +
 gcc/testsuite/g++.dg/modules/omp-2_b.C | 1 +
 5 files changed, 5 insertions(+)

diff --git a/gcc/testsuite/g++.dg/modules/omp-1_a.C b/gcc/testsuite/g++.dg/modules/omp-1_a.C
index 722720a0e93..94e1171f03c 100644
--- a/gcc/testsuite/g++.dg/modules/omp-1_a.C
+++ b/gcc/testsuite/g++.dg/modules/omp-1_a.C
@@ -1,4 +1,5 @@
 // { dg-additional-options "-fmodules-ts -fopenmp" }
+// { dg-require-effective-target pthread }
 
 export module foo;
 // { dg-module-cmi foo }
diff --git a/gcc/testsuite/g++.dg/modules/omp-1_b.C b/gcc/testsuite/g++.dg/modules/omp-1_b.C
index f3f5d92e517..09d97e4ac4e 100644
--- a/gcc/testsuite/g++.dg/modules/omp-1_b.C
+++ b/gcc/testsuite/g++.dg/modules/omp-1_b.C
@@ -1,4 +1,5 @@
 // { dg-additional-options "-fmodules-ts -fopenmp" }
+// { dg-require-effective-target pthread }
 
 import foo;
 
diff --git a/gcc/testsuite/g++.dg/modules/omp-1_c.C b/gcc/testsuite/g++.dg/modules/omp-1_c.C
index f30f6115277..599a5a5d34f 100644
--- a/gcc/testsuite/g++.dg/modules/omp-1_c.C
+++ b/gcc/testsuite/g++.dg/modules/omp-1_c.C
@@ -1,4 +1,5 @@
 // { dg-additional-options "-fmodules-ts" }
+// { dg-require-effective-target pthread }
 
 import foo;
 
diff --git a/gcc/testsuite/g++.dg/modules/omp-2_a.C b/gcc/testsuite/g++.dg/modules/omp-2_a.C
index d2291b6bbe0..b0d4bbc6e8a 100644
--- a/gcc/testsuite/g++.dg/modules/omp-2_a.C
+++ b/gcc/testsuite/g++.dg/modules/omp-2_a.C
@@ -1,4 +1,5 @@
 // { dg-additional-options "-fmodules-ts -fopenmp" }
+// { dg-require-effective-target pthread }
 
 export module foo;
 // { dg-module-cmi foo }
diff --git a/gcc/testsuite/g++.dg/modules/omp-2_b.C b/gcc/testsuite/g++.dg/modules/omp-2_b.C
index 39f34c70275..aeee4d1561a 100644
--- a/gcc/testsuite/g++.dg/modules/omp-2_b.C
+++ b/gcc/testsuite/g++.dg/modules/omp-2_b.C
@@ -1,4 +1,5 @@
 // { dg-additional-options "-fmodules-ts" }
+// { dg-require-effective-target pthread }
 
 import foo;
 
-- 
2.33.1

